swf/actors/worker.py

```diff
@@ -127,6 +127,9 @@ def heartbeat(self, task_token, details=None):
                 task_token,
                 format.heartbeat_details(details),
             )
+           # instrumentation
+           self.send_endpoint_usage("RecordActivityTaskHeartbeat")
```

Comment:
All the instrumentation lines should be put *before* the actual call to the endpoint. Otherwise if the call makes a 429, an exception is raised and you'll never hit influxdb for this call. Which means in practice that graphs won't ever reach the rate limiting thresholds, whereas we want them to do so. I'm actually unsure if when SWF answers a 429 it's counted in our rate limit, but I think we're safer if we have an exaggerated view than the other way around. Last but not least the line above is never called (it's after a "return").

Also: I'd be happy that we integrate instrumentation in a generic way that doesn't pollute API calls. Basically `connection.call_endpoint()` should result in the endpoint instrumentation being triggered AND the endpoint being called. But that can come in a second phase I think.